jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -486,6 +492,11 @@ List<Image> getCachedBaseImages()
       }
 
       ManifestTemplate manifest = Verify.verifyNotNull(manifestAndConfigFound.get().getManifest());
+      // Verify all layers described in manifest are present in cache
+      if (!baseImageLayersCache.allLayersCached(manifest)) {
+        return Collections.emptyList();
+      }
+
```

Comment:
It gets messier here to combine them I think? (Since grabbing the manifest is conditional on `manifestAndConfigFound.isPresent()`, plus it also needs to perform the `verifyNotNull` check)

Reply:
Ah I see! thanks for trying this out. Hm thinking of a way in which we could remove the number of if statements since they are all testing similar things but just at different levels of granularity. Is it possible to put them in a helper method with a descriptive name?

Reply:
Excellent refactoring!